### PR TITLE
refactor(blocks): code block toolbar

### PR DESCRIPTION
**Pull request by fundon**
Closes [BS-1287](https://linear.app/affine-design/issue/BS-1287/code-block-options-copy-link-to-block-button)

### CODE REVIEW

**Comment by fundon** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8077?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8077** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8077?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8072** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8072?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8070** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8070?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/affine/components/src/rich-text/inline/presets/nodes/reference-node/reference-popup.ts [hunk 1]
```diff
@@ -192,7 +193,7 @@ export class ReferencePopup extends WithDisposable(LitElement) {
             button => button.label,
             ({ label, icon, action, disabled }) => html`
               <editor-menu-action
-                ?aria-label=${label}
+                aria-label=${ifDefined(label)}
```
Comment (Saul-Mirone) [inline packages/affine/components/src/rich-text/inline/presets/nodes/reference-node/reference-popup.ts:196]: Just curious. What's the difference here? why `?aria-label` won't work?

Comment (fundon) [reply packages/affine/components/src/rich-text/inline/presets/nodes/reference-node/reference-popup.ts:196]: `?aria-label` for boolean values, if it is true, the `aria-label` property will be added, but has no specific value.

File: packages/blocks/src/root-block/widgets/code-toolbar/type.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  get doc() {
+    return this.blockComponent.doc;
+  }
+
+  get host() {
+    return this.blockComponent.host;
+  }
+
+  get selectedBlockModels() {
+    return [this.blockComponent.model];
```
Comment (fourdim) [inline packages/blocks/src/root-block/widgets/code-toolbar/type.ts:38]: Here.

**Review by fourdim — APPROVED** [review]

**Review by fourdim — APPROVED** [review]

**Pull request merged**